sn_interface/src/messaging/signature_aggregator.rs

```diff
@@ -71,90 +67,30 @@ impl SignatureAggregator {
         hasher.update(&public_key.to_bytes());
         hasher.finalize(&mut hash);
 
-        self.map
-            .entry(hash)
-            .or_insert_with(State::new)
-            .add(sig_share)
-            .map(|signature| SectionSig {
-                public_key,
-                signature,
-            })
-    }
-
-    fn remove_expired(&mut self) {
-        let expiration = self.expiration;
-        let mut to_remove = vec![];
-
-        for (digest, state) in &self.map {
-            if state.modified.elapsed() >= expiration {
-                to_remove.push(*digest);
-            }
-        }
-
-        self.map.retain(|digest, _| !to_remove.contains(digest))
-    }
-}
+        // save the sig share
+        let current_shares = self.map.entry(hash).or_insert(BTreeSet::new());
```

Comment:
Use `.or_default()` here, `.or_insert(BTreeSet::new())` will always allocate a new `BTreeSet` even when we already have an entry for the hash.
```suggestion
        let current_shares = self.map.entry(hash).or_default();
```